ouroboros-consensus/src/unstable-consensus-testlib/Test/Util/Serialisation/Roundtrip.hs

```diff
@@ -617,3 +626,46 @@ decodeThroughSerialised
 decodeThroughSerialised dec decSerialised = do
     serialised <- decSerialised
     fromSerialised dec serialised
+
+{------------------------------------------------------------------------------
+  Roundtrip tests for examples
+------------------------------------------------------------------------------}
+
+examplesRoundtrip ::
+     forall blk . (SerialiseDiskConstraints blk, Eq blk, Show blk, LedgerSupportsProtocol blk)
+  => CodecConfig blk
+  -> Examples blk
+  -> [TestTree]
+examplesRoundtrip codecConfig examples =
+  [ testRoundtripFor "Block"                 (encodeDisk codecConfig) (decodeDisk codecConfig)           exampleBlock
+  , testRoundtripFor "Header hash"           encode                   (const <$> decode)                 exampleHeaderHash
+  , testRoundtripFor "Ledger state"          (encodeDisk codecConfig) (const <$> decodeDisk codecConfig) exampleLedgerState
+  , testRoundtripFor "Annotated tip"         (encodeDisk codecConfig) (const <$> decodeDisk codecConfig) exampleAnnTip
+  , testRoundtripFor "Chain dependent state" (encodeDisk codecConfig) (const <$> decodeDisk codecConfig) exampleChainDepState
+  , testRoundtripFor "Extended ledger state" encodeExt                (const <$> decodeExt)              exampleExtLedgerState
+  ]
+  where
```

Comment:
I wonder if we can trade in some horizontal real estate for vertical real estate

Reply:
I tried putting the type in one line but I think it's too long. However, I corrected the indentation in `testRoundtripFor`.